CDTDatastoreTests/CDTReplicationTests.m

```diff
@@ -138,10 +175,41 @@ - (void)testURLCredsReplacedWithCookieInterceptorPush
     // check the underlying source to make sure it doesn't contain the userinfo
     // and check that the interceptors list contains the cookie interceptor.
     XCTAssertEqualObjects(@"http://example.com", push.target.absoluteString);
-    XCTAssertEqual(push.httpInterceptors.count, 1);
+    // 2 interceptors - because the 429 backoff interceptor is also present
+    XCTAssertEqual(push.httpInterceptors.count, 2);
     XCTAssertEqualObjects([push.httpInterceptors[0] class], [CDTSessionCookieInterceptor class]);
 }
 
+- (void)test429Retry
+{
+    NSError *error;
+    // simple remote to send 429
+    int port = 9999;
+    SimpleHttpServer *server = [[SimpleHttpServer alloc] initWithHeader:@"HTTP/1.0 429 Too Many Requests\r\n\r\n"
+                                                                   port:port];
+    [server start];
+    NSString *remoteUrl = [NSString stringWithFormat:@"http://127.0.0.1:%d", port];
+    
+    CDTDatastore *tmp = [self.factory datastoreNamed:@"test_database" error:&error];
+    CDTPullReplication *pull =
+    [CDTPullReplication replicationWithSource:[NSURL URLWithString:remoteUrl] target:tmp];
+    CDTReplicatorFactory *replicatorFactory =
+    [[CDTReplicatorFactory alloc] initWithDatastoreManager:self.factory];
+    
+    CDTReplicator *replicator = [replicatorFactory oneWay:pull error:&error];
+    
+    dispatch_group_t taskGroup = dispatch_group_create();
+    [replicator startWithTaskGroup:taskGroup error:&error];
+    
+    dispatch_group_wait(taskGroup, DISPATCH_TIME_FOREVER);
+
+    // after 10 retries the sleep time should equal 512:
+    // 250ms * (2^11)
+    XCTAssertEqual(512, ((CDTRequestLimitInterceptor*)(replicator.cdtReplication.httpInterceptors[0])).sleep);
```

Comment:
@ricellis good point, and since the test completes much faster, something must be broken somewhere